drivers/serial/uart_gecko.c

```diff
@@ -354,19 +449,81 @@ static const struct uart_driver_api uart_gecko_driver_api = {
 			gpioModePushPull, 1				       \
 	}
 
-#define GECKO_UART_PINS(idx)						       \
+#define GECKO_UART_RX_TX_PINS(idx)					       \
 	.pin_rx = PIN_UART_RXD(idx),					       \
 	.pin_tx = PIN_UART_TXD(idx),
 
+#ifdef UART_GECKO_HW_FLOW_CONTROL
+
+#ifdef CONFIG_SOC_GECKO_HAS_INDIVIDUAL_PIN_LOCATION
+#define GECKO_UART_RTS_CTS_PIN_LOCATIONS(idx)				       \
+	.loc_rts = COND_CODE_1(DT_INST_PROP(idx, hw_flow_control),	       \
+		   (DT_INST_PROP_BY_IDX(idx, location_rts, 0)),		       \
+		   (0)),						       \
+	.loc_cts = COND_CODE_1(DT_INST_PROP(idx, hw_flow_control),	       \
+		   (DT_INST_PROP_BY_IDX(idx, location_cts, 0)),		       \
+		   (0)),
+#define VALIDATE_GECKO_UART_RTS_CTS_PIN_LOCATIONS(idx)			       \
+	COND_CODE_1(DT_INST_PROP(idx, hw_flow_control),			       \
+		    (BUILD_ASSERT(DT_INST_NODE_HAS_PROP(idx, location_rts) &&  \
```

Comment:
This ensures that location-rts and location-cts are set in the dts, when hw-flow-control is enabled.
But there are a lot of errors displayed in that case. You have to scroll up to the first error to the the message "DTS location-rts and location-cts are mandatory".
Maybe there is a better way to do this with a cleaner error message.

Reply:
There isn't a better solution available today.  We'd need to look at adding some additional syntax to the binding to convey if prop X than require prop Y.

Reply:
Converted this into an issue #27483